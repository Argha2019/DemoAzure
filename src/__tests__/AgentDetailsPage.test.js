import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AgentDetailsTemplate from '../components/templates/AgentListTemplate/AgentDetailsTemplate.jsx';
import Cookies from 'js-cookie';

//test suite for AgentDetailsPage Search
describe('AgentDetailsPage Search Component', () => {

  //generate snapshot
   it('should render snapshot', () => {
      const wrapper = shallow(<AgentDetailsTemplate />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    //check DOM elements of AgentDetailsPage search
    it('should render required form elements', () => {
	 Cookies.set('adminData' , "Manjeet");
     const wrapper = shallow(<AgentDetailsTemplate />);
     const form = wrapper.find('.MyForm1');
     expect(form.length).toBe(1);
     expect(form.find('.RegisterButton').length).toBe(1);
     
  });
 
 //check the onChange function for right format of last name input
         it('onChange is called for country', () => {
		 	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);

          
           const event = {
             preventDefault: () => {},
             target: {
               name: "country",
               value: "India",
             }
           }

           wrapper.setState({
            edit: true,
			readOnly : false,

          });
          wrapper.update();
          const form = wrapper.find('#MyForm1');
             wrapper.find('#country1').simulate('change', event);
             expect(wrapper.state('country')).toEqual('India');
     
           });

          it('onChange is called for first name', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "firstName",
               value: "Kartikey",
             }
           }

           wrapper.setState({
            edit:true
          });
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#firstName1').simulate('change', event);
             expect(wrapper.state('firstName')).toEqual('Kartikey');
     
           });

          it('onChange is called for last name', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "lastName",
               value: "Ghai",
             }
           }

           wrapper.setState({
            edit:true
          });
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#lastName1').simulate('change', event);
             expect(wrapper.state('lastName')).toEqual('Ghai');
     
           });

          it('onChange is called for state', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "state",
               value: "Punjab",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#state1').simulate('change', event);
             expect(wrapper.state('state')).toEqual('Punjab');
     
           });

          it('onChange is called for phone', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "phone",
               value: "1234567890",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#phone1').simulate('change', event);
             expect(wrapper.state('phone')).toEqual('1234567890');
     
           });

          it('onChange is called for mail', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "mail",
               value: "abcd@gmail.com",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#mail1').simulate('change', event);
             expect(wrapper.state('mail')).toEqual('abcd@gmail.com');
           });

          it('onChange is called for addressLine1', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "addressLine1",
               value: "Gurgaon",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#addressLine11').simulate('change', event);
             expect(wrapper.state('addressLine1')).toEqual('Gurgaon');
           });

          it('onChange is called for addressLine2', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "addressLine2",
               value: "Gurgaon",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#addressLine21').simulate('change', event);
             expect(wrapper.state('addressLine2')).toEqual('Gurgaon');
           });

          it('onChange is called for pin code', () => {
		  	 Cookies.set('adminData' , "Manjeet");

           const wrapper = shallow(<AgentDetailsTemplate />);
          
           const event = {
             preventDefault: () => {},
             target: {
               name: "pincode",
               value: "123456",
             }
           }

           wrapper.setState({
            edit:true
          }); 
          wrapper.update();
          const form = wrapper.find('.MyForm1');
             wrapper.find('#pincode1').simulate('change', event);
             expect(wrapper.state('pincode')).toEqual('123456');
           });

         
			//cancel button
           it('cancelProfile is called', () => {
		   	 Cookies.set('adminData' , "Manjeet");

            const wrapper = shallow(<AgentDetailsTemplate/>);
           
            const event = {
              preventDefault: () => {},
            }
 
            wrapper.setState({
             edit:true
           });
           wrapper.update();
           const form = wrapper.find('.MyForm');
              wrapper.find('#cancel1').last().simulate('click', event);
              expect(wrapper.state('edit')).toEqual(false);
      
            });


			//Submit button
			it('submit button is called', () => {
		   	 Cookies.set('adminData' , "Manjeet");

            const wrapper = shallow(<AgentDetailsTemplate/>);
           
            const event = {
              preventDefault: () => {},
            }
 
            wrapper.setState({
             edit:true
           });
           wrapper.update();
           const form = wrapper.find('.MyForm');
              wrapper.find('#submit1').last().simulate('click', event);
              expect(wrapper.state('edit')).toEqual(false);
      
            });


			//Edit Button
			it('edit button is called', () => {
		   	 Cookies.set('adminData' , "Manjeet");

            const wrapper = shallow(<AgentDetailsTemplate/>);
           
            const event = {
              preventDefault: () => {},
            }
 
            wrapper.setState({
             edit:false
           });
           wrapper.update();
           const form = wrapper.find('.MyForm');
              wrapper.find('#edit1').last().simulate('click', event);
              expect(wrapper.state('edit')).toEqual(true);
      
            });
 
 });
