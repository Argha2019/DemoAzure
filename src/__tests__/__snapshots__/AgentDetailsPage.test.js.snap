// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentDetailsPage Search Component should render snapshot 1`] = `
<AgentTemplate>
  <div
    className="register-wrap"
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Agent Details Page
    </h1>
    <div
      className="Oms register-html"
    >
      <Form
        className="MyForm2 register-form"
        inline={false}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="Status"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              for="status"
              srOnly={false}
            >
              Status
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <ReactSwitch
              activeBoxShadow="0 0 2px 3px #3bf"
              boxShadow={null}
              checked={true}
              checkedIcon={false}
              disabled={true}
              height={28}
              id="status"
              offColor="#888"
              offHandleColor="#fff"
              onColor="#080"
              onHandleColor="#fff"
              uncheckedIcon={false}
              width={56}
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="Username"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              UserName
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              defaultValue=""
              name="username"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="role"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Role
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              name="role"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="firstName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              First Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="firstName"
              defaultValue=""
              name="firstName"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="lastName"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Last Name
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="lastName"
              defaultValue=""
              name="lastName"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="dob"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Date of Birth
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              defaultValue=""
              name="dob"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="phone"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Phone
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="phone"
              defaultValue=""
              name="phone"
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="mail"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Mail ID
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="mail"
              defaultValue=""
              name="mail"
              readOnly={true}
              required={true}
              type="email"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine1"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Address Line 1
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine1"
              defaultValue=""
              name="addressLine1"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="addressLine2"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Address Line 2
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="addressLine2"
              defaultValue=""
              name="addressLine2"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="state"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              State
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="state"
              defaultValue=""
              name="state"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="country"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Country
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="country"
              defaultValue=""
              name="country"
              readOnly={true}
              required={true}
              type="text"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="pincode"
        >
          <Col
            className="alignLabel"
            md={3}
            sm={4}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Pincode
            </FormLabel>
          </Col>
          <Col
            md={9}
            sm={8}
          >
            <FormControl
              className="pincode"
              name="pincode"
              readOnly={true}
              required={true}
              type="number"
            />
          </Col>
        </FormGroup>
      </Form>
    </div>
  </div>
</AgentTemplate>
`;
